fuse_constraints/include/fuse_constraints/marginal_constraint.h

```diff
@@ -47,20 +49,22 @@
 #include <boost/serialization/base_object.hpp>
 #include <boost/serialization/export.hpp>
 #include <boost/serialization/shared_ptr.hpp>
+#include <boost/serialization/unique_ptr.hpp>
 #include <boost/serialization/vector.hpp>
 #include <boost/tuple/tuple.hpp>
 #include <ceres/cost_function.h>
 
 #include <algorithm>
 #include <cassert>
+#include <functional>
+#include <iterator>
 #include <ostream>
 #include <string>
```

Comment:
```suggestion
#include <stdexcept>
#include <string>
```
required by `std::runtime_error`